app/frontend/wallet/shelley/shelley-transaction.ts

```diff
@@ -345,11 +345,11 @@ const cborizeTxVotingRegistration = ({
 
 const cborizeTxAuxiliaryVotingData = (
   txAuxiliaryData: TxAuxiliaryData,
-  signatureHex: string
+  signatureHex: string | undefined
```

Comment:
I see in the upstream logic you relaxed the type of the signatureHex parameter, but it actually should always be defined

Reply:
I allowed for this to be undefined because it is called with `auxiliaryDataSupplement.catalystSignature` as the value for `signatureHex` which can be undefined

https://github.com/trezor/connect/blob/2955b5ef92a3c77ccf3b1353f74f37fa4a2ff337/src/ts/types/networks/cardano.d.ts#L201

I can keep this type as defined and add a runtime check for where it is called with the undefined value

Reply:
as explained in https://github.com/vacuumlabs/adalite/pull/1192#discussion_r741223653 let's add a runtime check